core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java

```diff
@@ -239,13 +247,22 @@ public StateComputerLedger.StateComputerResult prepare(
                             .toList(),
                         v.getLedgerHeader().getAccumulatorState().getAccumulatorHash()))
             .toList();
+    var gapRoundLeaderKeys =
+        LongStream.range(roundDetails.previousQcRoundNumber() + 1, roundDetails.roundNumber())
+            .mapToObj(Round::of)
+            .map(this.currentProposerElection.get()::getProposer)
+            .map(BFTValidatorId::getKey)
```

Comment:
So theoretically it's possible for two validators to use the same key. Therefore we should actually use the validator component address here.